department-of-veterans-affairs/va.gov-team
# Update GitHub Pages index view to be Documentation Landing Page MVP

### Screenshot

![doc-landing-page-mvp](https://user-images.githubusercontent.com/6130520/77177451-83ab4d80-6a93-11ea-801e-0fab30dd0d4a.gif)

### Testing Instructions

1. Navigate to an HTML previewer (for example: https://htmlpreview.github.io/)
1. Paste the link to the HTML file into the HTML previewer
   - HTML link: https://github.com/department-of-veterans-affairs/va.gov-team/blob/doc-landing-page-v1/docs/index.html
1. Click **Preview**
1. Check out the landing page

### Ticket

https://app.zenhub.com/workspaces/vsp-5cedc9cce6e3335dc5a49fc4/issues/department-of-veterans-affairs/va.gov-team/6299

### Description

This PR updates the current `va.gov-team` GitHub Pages index view from an `index.md` to an `index.html`. The content in the `index.html` represents the MVP of our documentation landing page. 

Since this is an MVP, we wanted to focus on speed and function-over-form. I felt the fastest way to validate our idea would be to build a static HTML file. I attempted to use our Design System, but that resulted in me spending time worrying about dependencies/building/deploying, and where all of that code should live. 

I asked Megan for some direction, and she said it wasn't necessary to use the Design System for the MVP. 

Ultimately, I chose to use a utility-first CSS framework called [Tailwinds](https://tailwindcss.com/) so I could focus 100% on building the page. 

The following paragraph from [Tailwind's docs](https://tailwindcss.com/docs/utility-first/) succinctly explains why I went with a utility-first CSS framework:

> But once you’ve actually built something this way, you’ll quickly notice some really important benefits:
> 
> - **You aren’t wasting energy inventing class names.** No more adding silly class names like sidebar-inner-wrapper just to be able to style something, and no more agonizing over the perfect abstract name for something that’s really just a flex container.
> 
> - **Your CSS stops growing.** Using a traditional approach, your CSS files get bigger every time you add a new feature. With utilities, everything is reusable so you rarely need to write new CSS.
> 
> - **Making changes feels safer.** CSS is global and you never know what you’re breaking when you make a change. Classes in your HTML are local, so you can change them without worrying about something else breaking. - https://tailwindcss.com/docs/utility-first/

### Tiny URL (old content from `index.md` file until this PR gets merged)

https://tinyurl.com/vagovdocs

### Tradeoffs/Gotchas

- I asked Megan if we should incorporate Andrew's original content from the `index.md` file, and she said not to worry about it. 
- According to Jon W. on the Analytics team, [the link type `class`-es should be at start of `class` string](https://app.zenhub.com/workspaces/vsp-5cedc9cce6e3335dc5a49fc4/issues/department-of-veterans-affairs/va.gov-team/6354#issuecomment-601422757).
- To move as quickly as possible, I chose to use Tailwinds via a CDN. Unfortunately, [using Tailwinds via CDN](https://tailwindcss.com/docs/installation#using-tailwind-via-cdn) comes with the following tradeoffs:
  - You can't customize [Tailwind's default theme](https://tailwindcss.com/docs/customizing-colors#default-color-palette)
  - You can't use any [directives](https://tailwindcss.com/docs/functions-and-directives/) like `@apply`, `@variants`, etc.
  - You can't enable features like [`group-hover`](https://tailwindcss.com/docs/pseudo-class-variants/#group-hover)
  - You can't install third-party plugins
